### PR TITLE
feat(config): support `.nuxtrc`

### PR DESCRIPTION
**Pull request by pi0**
## Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

Support `.nuxtrc` (per project and user home) to extend `nuxt.config` without making modifications. The use case is for any module that wants to persist a state or config per project on an automated basis like telemetry module to set `telemetry.enabled`. RC format is flattened INI but config is read unflattened with native types (see [rc9](https://github.com/nuxt-contrib/rc9))

Loading priority: configOverrides > nuxtConfig > .nuxtrc > .nuxtrc (global)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [x] I have added tests to cover my changes (if not applicable, please state why)
- [ ] All new and existing tests are passing.

### CODE REVIEW

**Review by clarkdo — COMMENTED** [review]
The change looks good.

I'm wondering if it's doing different things like what most framworks are doing, like babel won't have `babel.config.js --falback-to-->  .babelrc`, eslint wont have `.eslintrc.js --falback-to--> .eslintrc`.

Another point is : Regarding to use rc file, is there anything impossible or hard to be implemented by using `nuxt.config.js` (or maybe a global `nuxt.config.js`) ?

**Comment by pi0** [comment]
> I'm wondering if it's doing different things like what most framworks are doing, like babel won't have babel.config.js --falback-to--> .babelrc, eslint wont have .eslintrc.js --falback-to--> .eslintrc.

Indeed this is same behavior we will have. The current loadConfig fixture for nuxtrc does not have a `nuxtconfig.js` file.  Main different with other (js) frameworks, is that nuxtrc is flat INI that allows easy modifications by automated tools (even a bash script) but loaded as valid json.

> Another point is : Regarding to use rc file, is there anything impossible or hard to be implemented by using nuxt.config.js (or maybe a global nuxt.config.js) ?

The most advantage is updating as for `nuxt.config` we need to parse syntax with different possibilities ({typescript, esm, cjs} x {exported object, exported function}) compared to a dead-simple syntax and using `rc.update()`. It also makes `nuxt.config` more readable by hiding tokens or configs like telemetry into a plain file. Another advantage is that this file can be optionally checked-in into a repository so based on project requirements can be gitignored or not.

**Comment by clarkdo** [comment]
So if we always respect `nuxt.config.js`, `.nuxtrc` and `~/.nuxtrc` at same time, do we need to provide a way to at least disable `~/.nuxtrc` in an app for edge cases ?

**Comment by pi0** [comment]
> So if we always respect nuxt.config.js, .nuxtrc and ~/.nuxtrc at same time, do we need to provide a way to at least disable ~/.nuxtrc in an app for edge cases ?

Agree. If we find a valid problematic case can support CLI flag but I don't think it would be necessary for initial support.

File: packages/config/src/load.js [hunk 1]
```diff
@@ -87,6 +88,15 @@ export async function loadNuxtConfig ({
     options.rootDir = rootDir
   }
 
+  // Load Combine configs
+  // Periority: configOverrides > nuxtConfig > .nuxtrc > .nuxtrc (global)
```
Comment (Atinux) [inline packages/config/src/load.js:92]: Priority*

**Review by atinux — DISMISSED** [review]

File: packages/config/src/load.js (hunk 1 above)
Comment (pi0) [reply packages/config/src/load.js]: Fixed

**Pull request merged**